test/parser.js

```diff
@@ -0,0 +1,77 @@
+import { expect } from 'chai';
+import fury from 'fury';
+import Parser from '../src/parser';
+
+describe('Parser', () => {
+  let parser;
+
+  before(() => {
+    parser = new Parser({ minim: fury.minim });
+    parser.swagger = {
+      consumes: [],
+      produces: [],
+    };
+  });
+
+  context('content types', () => {
+    it('gathers null response type when no examples or produces', () => {
+      const methodValue = {};
+      const examples = {};
+      const contentTypes = parser.gatherResponseContentTypes(methodValue, examples);
+
+      expect(contentTypes).to.deep.equal([null]);
+    });
+
+    it('gathers all example response content types', () => {
+      const methodValue = {};
+      const examples = {
+        'application/json': '',
+        'application/hal+json': '',
+        'application/xml': '',
+      };
+      const contentTypes = parser.gatherResponseContentTypes(methodValue, examples);
+
+      expect(contentTypes).to.deep.equal([
+        'application/json',
+        'application/hal+json',
+        'application/xml',
+      ]);
+    });
+
+    it('gathers first JSON produces without examples', () => {
+      const methodValue = {
+        produces: [
+          'application/json',
+          'application/hal+json',
+        ],
+      };
+      const examples = {};
+      const contentTypes = parser.gatherResponseContentTypes(methodValue, examples);
+
+      expect(contentTypes).to.deep.equal(['application/json']);
+    });
+
+    it('gathers only example response content types with produces', () => {
+      const methodValue = {
+        produces: [
+          'application/json',
+          'application/problem+json',
+        ],
+      };
+      const examples = {
+        'application/vnd.error+json': '',
+      };
+      const contentTypes = parser.gatherResponseContentTypes(methodValue, examples);
+
+      expect(contentTypes).to.deep.equal(['application/vnd.error+json']);
+    });
+
+    it('rejects invalid content types when gathering response content types', () => {
```

Comment:
Should this add a warning about invalid content type in the API description? 🤔

Reply:
That's already handled elsewhere. It is not handled in this specific unit.

- https://github.com/apiaryio/fury-adapter-swagger/blob/master/src/parser.js#L178
- https://github.com/apiaryio/fury-adapter-swagger/blob/master/src/parser.js#L712